src/main/java/javaslang/collection/Vector.java

```diff
@@ -765,12 +773,20 @@ public int length() {
 
     @Override
     public Vector<T> prepend(T element) {
-        return insert(0, element);
+        final int newIndexShift = indexShift - 1;
```

Comment:
Can this be < 0? is the indexShift always >= 0? I have to read further through the code...